crates/bevy_feathers/src/controls/toggle_switch.rs

```diff
@@ -0,0 +1,247 @@
+            slide_ent,
+            disabled,
+            checked,
+            hovered.0,
+            outline_bg,
+            outline_border,
+            slide_style,
+            slide_color,
+            &mut commands,
+        );
+    }
+}
+
+fn update_switch_styles_remove(
+    q_switches: Query<
+        (
+            Entity,
+            Has<InteractionDisabled>,
+            Has<Checked>,
+            &Hovered,
+            &ThemeBackgroundColor,
+            &ThemeBorderColor,
+        ),
+        With<ToggleSwitchOutline>,
+    >,
+    q_children: Query<&Children>,
+    mut q_slide: Query<(&mut Node, &ThemeBackgroundColor), With<ToggleSwitchSlide>>,
+    mut removed_disabled: RemovedComponents<InteractionDisabled>,
+    mut removed_checked: RemovedComponents<Checked>,
+    mut commands: Commands,
+) {
+    removed_disabled
+        .read()
+        .chain(removed_checked.read())
+        .for_each(|ent| {
+            if let Ok((switch_ent, disabled, checked, hovered, outline_bg, outline_border)) =
+                q_switches.get(ent)
+            {
+                let Some(slide_ent) = q_children
+                    .iter_descendants(switch_ent)
+                    .find(|en| q_slide.contains(*en))
+                else {
+                    return;
+                };
+                let (ref mut slide_style, slide_color) = q_slide.get_mut(slide_ent).unwrap();
+                set_switch_colors(
+                    switch_ent,
+                    slide_ent,
+                    disabled,
+                    checked,
+                    hovered.0,
+                    outline_bg,
+                    outline_border,
+                    slide_style,
+                    slide_color,
+                    &mut commands,
+                );
+            }
+        });
+}
+
+fn set_switch_colors(
+    switch_ent: Entity,
+    slide_ent: Entity,
+    disabled: bool,
+    checked: bool,
+    hovered: bool,
+    outline_bg: &ThemeBackgroundColor,
+    outline_border: &ThemeBorderColor,
+    slide_style: &mut Mut<Node>,
+    slide_color: &ThemeBackgroundColor,
+    commands: &mut Commands,
+) {
+    let outline_border_token = match (disabled, hovered) {
+        (true, _) => tokens::SWITCH_BORDER_DISABLED,
+        (false, true) => tokens::SWITCH_BORDER_HOVER,
+        _ => tokens::SWITCH_BORDER,
+    };
+
+    let outline_bg_token = match (disabled, checked) {
+        (true, true) => tokens::SWITCH_BG_CHECKED_DISABLED,
+        (true, false) => tokens::SWITCH_BG_DISABLED,
+        (false, true) => tokens::SWITCH_BG_CHECKED,
+        (false, false) => tokens::SWITCH_BG,
+    };
+
+    let slide_token = match disabled {
+        true => tokens::SWITCH_SLIDE_DISABLED,
+        false => tokens::SWITCH_SLIDE,
+    };
+
+    let slide_pos = match checked {
+        true => Val::Percent(50.),
+        false => Val::Percent(0.),
+    };
+
+    // Change outline background
+    if outline_bg.0 != outline_bg_token {
+        commands
```

Comment:
I don't understand why we're mutating by commands here, and by &mut elsewhere. Can't we just make the caller pass us a `&mut ThemeBackgroundColor`?

If we're worried it might not exist, we could add it to the required components at runtime.